avogadro/mainwindow.cpp

```diff
@@ -1252,14 +1252,22 @@ void MainWindow::activeMoleculeEdited()
   if (!m_undo || !m_redo)
     return;
   if (m_molecule) {
-    if (m_molecule->undoMolecule()->undoStack().canUndo())
+    if (m_molecule->undoMolecule()->undoStack().canUndo()) {
       m_undo->setEnabled(true);
-    else
+      m_undo->setText(tr("&Undo ") + m_molecule->undoMolecule()->undoStack().undoText());
+    }
+    else {
       m_undo->setEnabled(false);
-    if (m_molecule->undoMolecule()->undoStack().canRedo())
+      m_undo->setText(tr("&Undo"));
+    }
+    if (m_molecule->undoMolecule()->undoStack().canRedo()) {
       m_redo->setEnabled(true);
-    else
+      m_redo->setText(tr("&Redo ") + m_molecule->undoMolecule()->undoStack().redoText());
```

Comment:
I'm a bit worried about this in terms of translation / localization. It assumes that "Redo X" is the translatable item, where I can imagine in some cases, the string might be the equivalent of "X Redo". Can we check with other Qt projects to see what to do here?

My guess is that we want something like tr("&Redo %1").arg(.. redoText()) to allow the order to be tuned by translators.